### PR TITLE
Extracts common interface for blob change handlers

### PR DESCRIPTION
**Pull request by cschierle**
Needs changes from #1549

Fixes: [OX-9285](https://scireum.myjetbrains.com/youtrack/issue/OX-9285)

### CODE REVIEW

File: src/main/java/sirius/biz/storage/layer2/BlobConvertibilityChangedHandler.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import sirius.kernel.di.std.Priorized;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Defines handlers to process {@link Blob blobs} after their {@link Blob#isInconvertible() convertibility status} changed.
+ *
+ * @see ProcessBlobChangesLoop
+ */
+public interface BlobConvertibilityChangedHandler extends Priorized {
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/BlobConvertibilityChangedHandler.java:20]: extract a super-interface for all BlobXXXChangedHandlers?

File: src/main/java/sirius/biz/storage/layer2/BlobConvertibilityChangedHandler.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package sirius.biz.storage.layer2;
+
+import sirius.kernel.di.std.Priorized;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Defines handlers to process {@link Blob blobs} after their {@link Blob#isInconvertible() convertibility status} changed.
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/BlobConvertibilityChangedHandler.java:16]: add a remark that handlers must be registered....

File: src/main/java/sirius/biz/storage/layer2/jdbc/SQLBlob.java [hunk 3]
```diff
@@ -208,6 +208,12 @@ public class SQLBlob extends SQLEntity implements Blob, OptimisticCreate {
     public static final Mapping INCONVERTIBLE = Mapping.named("inconvertible");
     private boolean inconvertible;
 
+    /**
+     * Stores if the convertibility of the blob changed, e.g. after updating the physical file.
+     */
```
Comment (andyHa) [inline src/main/java/sirius/biz/storage/layer2/jdbc/SQLBlob.java:213]:
```suggestion
     * Stores if the convertibility of the blob changed, e.g. after updating the physical file.
     * <p>
     * We store an extra flag here, which is internally toggled and then propagated via the
     * <tt>ProcessBlobChangesLoop</tt>
     * 
     */
```

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/storage/layer2/jdbc/SQLBlob.java [hunk 4]
```diff
@@ -239,13 +245,18 @@ protected void beforeSave() {
             createdOrRenamed = true;
         }
 
+        if (isNew() || isChanged(CONVERTIBILITY_CHANGED)) {
```
Comment (sabieber) [inline src/main/java/sirius/biz/storage/layer2/jdbc/SQLBlob.java:248]: Will this not result in the field being set back to true when it is set to false via `mango.update`?

File: src/main/java/sirius/biz/storage/layer2/mongo/MongoBlob.java [hunk 5]
```diff
@@ -246,13 +252,18 @@ protected void beforeSave() {
             createdOrRenamed = true;
         }
 
+        if (isNew() || isChanged(CONVERTIBILITY_CHANGED)) {
```
Comment (sabieber) [inline src/main/java/sirius/biz/storage/layer2/mongo/MongoBlob.java:255]: s.a.

**Comment by cschierle** [comment]
Replaced by https://github.com/scireum/sirius-biz/pull/1562

**Pull request closed without merging**